components/builder-db/src/pool.rs

```diff
@@ -98,13 +82,7 @@ impl Pool {
 
         let shard_id = match optional_shard_id {
             Some(id) => (id % SHARD_COUNT as u64) as u32,
-            None => {
-                let mut rng = rand::thread_rng();
-                match rng.choose(&self.shards) {
-                    Some(shard) => *shard,
-                    None => 0,
-                }
-            }
```

Comment:
Since get_shard is basically a no-op now, the hash computation and most of this function can actually go away as well... so the entire function is "self.get_shard(0)".   Since that's the case, it looks like we can safely get rid of the get_shard and get_raw functions altogether, and replace them with the Pool get function that can now just return the inner connection..

Reply:
Will do